pdc/settings_test.py

```diff
@@ -50,3 +50,45 @@
 COMPONENT_BRANCH_NAME_BLACKLIST_REGEX = r'^epel\d+$'
 DISABLE_RESOURCE_PERMISSION_CHECK = True
 SKIP_RESOURCE_CREATION = True
+
+
+class TestListener(object):
+    """
+    Context manager for messages from TestMessenger.
+    """
+    def __init__(self, messenger):
+        self.messenger = messenger
+        self.messages = []
+
+    def __enter__(self):
+        self.messenger.listeners.append(self)
+        return self.messages
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.messenger.listeners.remove(self)
+
+
+class TestMessenger(object):
+    """
+    Processes messages from message bus for tests.
+
+    ::
+
+        with messenger.listen() as messages:
+            # Make requests ...
+            topic, msg = messages[0]
+    """
+    def __init__(self):
+        self.listeners = []
+
+    def send_message(self, topic, msg):
+        for listener in self.listeners:
+            listener.messages.append((topic, msg))
+
+    def listen(self):
+        return TestListener(self)
```

Comment:
Should these two classes really be in the settings file? Maybe a comment explaining why would be useful.

Reply:
I could add special value `"test"` for the `settings.MESSAGE_BUS['MLP']` and move the classes to `pdc/apps/utils/messaging.py`. That way I don't have to modify how the global `messenger` variable is initiated.